### ISSUE TITLE
rustfmt-nightly crashes on Windows due to missing DLL

### ISSUE THREAD

**Issue by dthul** [rust-lang-nursery/rustfmt#1736]
My toolchain is nightly-x86_64-pc-windows-msvc (rustc 1.19.0-nightly).
I installed rustfmt using `cargo install rustfmt-nightly`.

Running `rustfmt` crashes with:

    The program can't start because syntax-e5250b2906338489.dll is missing from your computer.

`cargo fmt` works though.

**Comment by kraai** [comment]
I see a different, but possibly related, error when I run `rustfmt` from a Git Bash window:

```
.../.cargo/bin/rustfmt.exe: error while loading shared libraries: std-7c926edf41e020d3.dll: cannot open shared object file: No such file or directory
```

`cargo fmt` works.

I can reproduce the error by compiling and running the following program with `rustc`:

```rust
#![feature(rustc_private)]
extern crate syntax;
fn main() {}
```

**Comment by akhilpai** [comment]
I get similar errors with nightly-x86_64-pc-windows-msvc (rustc 1.20.0-nightly):

The code execution cannot proceed because rustc_errors-61cad14d0691e95.dll was not found. Reinstalling the Program may fix this problem.
The code execution cannot proceed because syntax_pos-3d23fc138472b069.dll was not found. Reinstalling the Program may fix this problem.
The code execution cannot proceed because syntax-8440894e45de05ed.dll was not found. Reinstalling the Program may fix this problem.
The code execution cannot proceed because std-e41dbd03022c4dff.dll was not found. Reinstalling the Program may fix this problem.

The relevant dlls live in:

.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\lib

**Comment by nrc** [comment]
I think you need to point to the DLLs some how (i.e., the windows equivalent of the last [tip](https://github.com/rust-lang-nursery/rustfmt#tips). I'm sorry I'm not sure how to do that on Windows.

**Comment by xftroxgpx** [comment]
I'm getting this too but on Linux:

```bash
$ /home/xftroxgpx/.cargo/bin/rustfmt
/home/xftroxgpx/.cargo/bin/rustfmt: error while loading shared libraries: libsyntax-8b48272fc0b24d10.so: cannot open shared object file: No such file or directory
$ ./target/debug/rustfmt 
./target/debug/rustfmt: error while loading shared libraries: libsyntax-8b48272fc0b24d10.so: cannot open shared object file: No such file or directory
```
The former is after a `cargo install rustfmt-nightly`, the latter is a git clone, cargo build.

Also tried:

```bash
$ RUST_SRC_PATH= PATH= ./target/debug/rustfmt 
./target/debug/rustfmt: error while loading shared libraries: libsyntax-8b48272fc0b24d10.so: cannot open shared object file: No such file or directory
$ RUST_SRC_PATH= PATH= /home/xftroxgpx/.cargo/bin/rustfmt 
/home/xftroxgpx/.cargo/bin/rustfmt: error while loading shared libraries: libsyntax-8b48272fc0b24d10.so: cannot open shared object file: No such file or directory

```

```
$ rustc -vV
!! executing '/home/xftroxgpx/.cargo/bin/rustc' with args: '-vV'
rustc 1.20.0-nightly (2652ce677 2017-07-17)
binary: rustc
commit-hash: 2652ce6771b114189cdc1f9bd026a10af0a714e0
commit-date: 2017-07-17
host: x86_64-unknown-linux-gnu
release: 1.20.0-nightly
LLVM version: 4.0

$ cargo -vV
!! executing '/home/xftroxgpx/.cargo/bin/cargo' with args: '-vV'
cargo 0.21.0-nightly (f709c35a3 2017-07-13)
release: 0.21.0
commit-hash: f709c35a35687b2665eb290695825375be91410b
commit-date: 2017-07-13
```

**Comment by nrc** [comment]
I think this is fixed by #1876
